### PR TITLE
Remove boot/ocamldep

### PR DESCRIPTION
**Pull request by nojb**
Following up to #1078, this PR removes `boot/ocamldep`, saving 2M in each bootstrap.

In fact `boot/ocamldep` is not used at all during building/bootstrapping/etc so apart from a couple trivial changes (see diff) no adaptations are necessary.

Does this need a Changes entry?

### CODE REVIEW

**Comment by xavierleroy** [comment]
I vaguely remember that ocamldep was added to boot/ so that dependencies could be rebuilt even if they are so broken that `tools/ocamldep` cannot be built.  @damiendoligez what do you think?  Is this OK or should we also replace uses of `tools/ocamldep` by `boot/ocamlc -depend` in various Makefiles?

**Comment by alainfrisch** [comment]
The goal of #1078 was precisely to avoid the need to commit an additional bytecode executable in `boot/`.  So yes, I think we should use `boot/ocamlc -depend` in Makefiles, and then get rid of `boot/ocamldep` for good.

**Comment by alainfrisch** [comment]
(Well, Makefiles currently refer to tools/ocamldep, not boot/ocamldep.)

**Comment by nojb** [comment]
Indeed, the Makefiles do not refer to `boot/ocamldep` at all, so the question is: should the references to `tools/ocamldep` be changed to `boot/ocamlc -depend` ?

**Comment by xavierleroy** [comment]
>should the references to tools/ocamldep be changed to boot/ocamlc -depend ?

If you do make sure it is `boot/ocamlrun boot/ocamlc -depend`.  There is some confusion between `boot/ocamlrun` and `byterun/ocamlrun` in the Makefiles presently, which will need to be fixed, so let's not add any.

The question has no easy answer.  
* In favor of `boot/ocamlc -depend`, it makes it possible to rebuild the Caml dependencies almost from a bare checkout: all you need is the initial build of `ocamlrun`.  So that's very robust against bad days.  
* Against the change: if ocamldep is improved or gets a bug fix, one full bootstrap cycle will be necessary until the changes propagate to `boot/ocamlc` and can be taken advantage of to rebuild the dependencies of the core system.

@damiendoligez any opinion?

**Comment by damiendoligez** [comment]
I'm completely in favor of removing `boot/ocamldep`, thus completing the work started by @lefessan, it will save a lot of bandwidth and disk space when cloning the repo.

@nojb: Please make sure that you can rebuild the dependencies from a fresh checkout without compiling any OCaml code.

**Comment by nojb** [comment]
If one comments out
```
beforedepend:: asmcomp/emit.ml
```
and
```
beforedepend:: bytecomp/opcodes.ml
```
from the main `Makefile` one can do `make depend` successfully from a fresh checkout (after compiling `ocamlyacc` and `ocamlrun`).

If we want to be able to do `make alldepend` from a fresh checkout we need to replace `tools/ocamldep` by `boot/ocamlc -depend` in the following places:

- `stdlib/Makefile`
- `lex/Makefile`
- `ocamltest/Makefile`
- `tools/Makefile`
- `ocamldoc/Makefile`
- `otherlibs/...`

**Comment by nojb** [comment]
@damiendoligez I checked that one could do `make depend` from a fresh checkout. One needs to comment out the lines in `Makefile` corresponding to files generated by OCaml tools but this is the same as before this PR.

We could replace `tools/ocamldep` by `boot/ocamldep -depend` everywhere else in the distribution which would allow us to do a `make alldepend` directly from a fresh checkout, but with the downside noted by Xavier.

Do you have an opinion on this?

**Comment by nojb** [comment]
Ping @damiendoligez.

File: Changes [hunk 1]
```diff
@@ -165,6 +165,10 @@ Working version
 - GPR#1585: optimize output of "ocamllex -ml"
   (Alain Frisch, review by Frédéric Bour and Gabriel Scherer)
 
+- GPR#1537: boot/ocamldep is no longer included in the distribution;
```
Comment (damiendoligez) [inline Changes]: Could you add "in the **source** distribution" to make it clear that binary distributions are not affected by this change?

**Review by damiendoligez — APPROVED** [review]

**Comment by damiendoligez** [comment]
Looks good except for the wording of the Changes entry. @nojb could you change it and rebase? Thanks.

File: Changes (hunk 1 above)
Comment (nojb) [reply Changes]: Fixed, thanks.

**Comment by nojb** [comment]
Will merge this by the end of the day unless someone speaks up.

**Pull request merged**